ctapipe_io_lst/__init__.py

```diff
@@ -51,15 +69,36 @@ def __init__(self, **kwargs):
         # To overcome this we substitute the input_url with first file matching
         # the specified file mask (copied from  MAGICEventSourceROOT).
 
-        if 'input_url' in kwargs.keys():
-            self.file_list = glob.glob(kwargs['input_url'])
-            self.file_list.sort()
-            kwargs['input_url'] = self.file_list[0]
+
+        super().__init__(**kwargs)
+
+        if self.multi_streams:
+            # test how many streams are there:
+            # file name must be [stream name]Run[all the rest]
+            # All the files with the same [all the rest] are opened
+
+            if '/' in self.input_url:
+                dir, name = self.input_url.rsplit('/', 1)
```

Comment:
Please us `os.path.split`  here

Reply:
It will also work  if there is no `/` in the path. Also you could combine it with `os.path.abspath`, so you don't need any case differentiation here.

```
directory, filename = os.path.split(os.path.abspath(self.input_url))
```